syntax = "proto3";

import "engine.proto";

package com.wazuh.api.engine.geo;

/* Entry of a db in the geo manager */
message DbEntry
{
    string name = 1;
    string path = 2;
    string type = 3;
}

/***************************************************
 * Add a mmddb to the geo manager, the db must be in the specified path
 *
 * command: geo.db/post (<resource>/<action>)
 **************************************************/
message DbPost_Request
{
    string path = 1;
    string type = 2;
}
// message DbPost_Response -> Return a GenericStatus_Response

/***************************************************
 * Remove a mmddb from the geo manager
 *
 * command: geo.db/delete (<resource>/<action>)
 **************************************************/
message DbDelete_Request
{
    string path = 1;
}
// message DbDelete_Response -> Return a GenericStatus_Response

/***************************************************
 * Get the list of mmddb used by the geo manager
 *
 * command: geo.db/list (<resource>/<action>)
 **************************************************/
message DbList_Request {}

message DbList_Response
{
    ReturnStatus status = 1;   // Status of the query
    optional string error = 2; // Error message if status is ERROR
    repeated DbEntry entries = 3;
}

/***************************************************
 * Remote upsert a mmddb to the geo manager, it downloads the db at the specified path
 *
 * command: geo.db/remoteUpsert (<resource>/<action>)
 **************************************************/
message DbRemoteUpsert_Request
{
    string path = 1;
    string type = 2;
    string dbUrl = 3;
    string hashUrl = 4;
}
// message DbRemoteUpsert_Response -> Return a GenericStatus_Response
